src/DssDirectDepositHub.t.sol

```diff
@@ -0,0 +1,863 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Copyright (C) 2021-2022 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity 0.6.12;
+
+import "ds-test/test.sol";
+import "./tests/interfaces/interfaces.sol";
+
+import {DssDirectDepositHub, D3MPoolLike, D3MPlanLike} from "./DssDirectDepositHub.sol";
+
+import {D3MTestPool} from "./tests/stubs/D3MTestPool.sol";
+import {D3MTestPlan} from "./tests/stubs/D3MTestPlan.sol";
+import {D3MTestGem} from "./tests/stubs/D3MTestGem.sol";
+import {D3MTestRewards} from "./tests/stubs/D3MTestRewards.sol";
+import {ValueStub} from "./tests/stubs/ValueStub.sol";
+
+interface Hevm {
+    function warp(uint256) external;
+
+    function store(
+        address,
+        bytes32,
+        bytes32
+    ) external;
+
+    function load(address, bytes32) external view returns (bytes32);
+}
+
+contract DssDirectDepositHubTest is DSTest {
+    uint256 constant WAD = 10**18;
+    uint256 constant RAY = 10**27;
+    uint256 constant RAD = 10**45;
+
+    Hevm hevm;
+
+    VatLike vat;
+    EndLike end;
+    D3MTestRewards rewardsClaimer;
+    DaiLike dai;
+    DaiJoinLike daiJoin;
+    D3MTestGem testGem;
+    TokenLike testReward;
+    SpotLike spot;
+    TokenLike weth;
+    address vow;
+    address pauseProxy;
+
+    bytes32 constant ilk = "DD-DAI-TEST";
+    DssDirectDepositHub directDepositHub;
+    D3MTestPool d3mTestPool;
+    D3MTestPlan d3mTestPlan;
+    ValueStub pip;
+
+    function setUp() public {
+        hevm = Hevm(
+            address(bytes20(uint160(uint256(keccak256("hevm cheat code")))))
+        );
+
+        vat = VatLike(0x35D1b3F3D7966A1DFe207aa4514C12a259A0492B);
+        end = EndLike(0xBB856d1742fD182a90239D7AE85706C2FE4e5922);
```

Comment:
reminder: update `end` contract in hub tests once [spell](https://etherscan.io/address/0x6febea3673AdF3925A01a61E46924B5b67A69Ec3) is cast.
```suggestion
        end = EndLike(0x0e2e8F1D1326A4B9633D96222Ce399c708B19c28);
```

Reply:
spell have been [cast](https://etherscan.io/tx/0xd891445bf62cb9df39a07be390f2650be820d4cb2c8efe51b4506e1763435f97)